apps/condo/domains/resident/schema/Resident.js

```diff
@@ -250,6 +252,25 @@ const Resident = new GQLListSchema('Resident', {
             if (!isNull(changedPropertyId) || !isNull(changedUnitName) || !isNull(changedUnitType)) {
                 await manageResidentToTicketClientConnections.delay(propertyId, unitType, unitName, userId, dv, sender)
             }
+
+            // handle soft delete
+            // in order to soft delete recurrent payment contexts
+            if (operation === 'update' && deletedAt) {
+                // remove RecurrentPaymentContext
+                const recurrentContexts = await RecurrentPaymentContext.getAll(context, {
+                    serviceConsumer: {
+                        resident: {
+                            id: updatedItem.id,
+                        },
+                    },
+                })
+
+                for (const recurrentContext of recurrentContexts) {
+                    if (!recurrentContext.deletedAt) {
```

Comment:
it's better to add deletedAt as filter condition an line 262

Reply:
I have extracted function for delete orphans recurrentContexts into acquiring/utils/serverSchema/helpers.js
The function includes this change